src/MCPClient/lib/clientScripts/verify_checksum.py

```diff
@@ -0,0 +1,276 @@
+            raise NoHashCommandAvailable()
+
+    def _call(self, *args):
+        """Make the call to Python subprocess and record the command being
+        called.
+        """
+        self.command_called = (self.COMMAND,) + args
+        return self._decode(subprocess.check_output(self.command_called))
+
+    def count_and_compare_lines(self, hashfile, objectsdir):
+        """Count the number of lines in a checksum file and compare with the
+        number of objects being transferred. The requirement of hashsum as
+        this microservice job is written is that the mapping is 1:1. There
+        isn't space for an empty line at the end of the file.
+        """
+        lines = self._count_lines(hashfile)
+        objects = self._count_files(objectsdir)
+        if lines == objects:
+            return True
+        self.job.pyprint(
+            "{}: Comparison failed with {} checksum lines and {} "
+            "transfer files".format(self.get_ext(hashfile), lines, objects),
+            file=sys.stderr,
+        )
+        return False
+
+    def compare_hashes(self, hashfile, objectsdir):
+        """Compare transfer files with the checksum file provided."""
+        if not self.count_and_compare_lines(hashfile, objectsdir):
+            return 1
+        try:
+            self._call("-c", "--strict", hashfile)
+            return 0
+        except subprocess.CalledProcessError as err:
+            if self.EXIT_NON_ZERO in str(err):
+                warn = "{}: comparison exited with status: {}. Please check the formatting of the checksums or integrity of the files.".format(
+                    self.get_ext(hashfile), err.returncode
+                )
+                self.job.pyprint(warn, file=sys.stderr)
+            for line in self._decode(err.output):
+                if not line:
+                    continue
+                if line.endswith(self.OKAY_STRING):
+                    continue
+                if (
+                    line.endswith(self.FAIL_STRING)
+                    or self.ZERO_STRING in line
+                    or self.IMPROPER_STRING in line
+                ):
+                    self.job.pyprint(
+                        "{}: {}".format(self.get_ext(hashfile), line), file=sys.stderr
+                    )
+                if line.endswith(self.FAILED_OPEN):
+                    self.job.pyprint(
+                        "{}: {}".format(self.get_ext(hashfile), line), file=sys.stderr
+                    )
+            return err.returncode
+
+    def version(self):
+        """Return version information for the command being called."""
+        try:
+            return self._call("--version")[0]
+        except subprocess.CalledProcessError:
+            return self.COMMAND
+
+    def get_command_detail(self):
+        """Provide some way for the user to get information out of the class to
+        write METS/PREMIS provenance information.
+        """
+        if not self.command_called:
```

Comment:
What's the code path for this condition to succeed? I noticed `get_command_detail` is only called after checking the result status is 0 so the command should have been saved.

Reply:
It's just gold-plating. We've given the caller an object to work with and one of the things the caller can do with that object is call this function independently of doing any actual hash comparison and so `self.command_called` will still be `None`. I've added context to the exception, and don't believe it's doing too much harm. But I can remove it?